hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -461,6 +462,14 @@ private void appendClientPortToCallerContextIfAbsent() {
               .append(CallerContext.CLIENT_PORT_STR, String.valueOf(Server.getRemotePort()))
               .setSignature(origSignature)
               .build());
+    } else {
+      String origContext = ctx == null ? null : ctx.getContext();
+      byte[] origSignature = ctx == null ? null : ctx.getSignature();
```

Comment:
If `ctx` was null, we wouldn't be here because `isClientPortInfoAbsent(ctx)` would have returned true already.

Reply:
if ctx was null , it should not add dfsrouterPort because the rpc call is directory connect namenode.  
When a rpc call is form dfsrouter, it will call RouterRpcClient.addClientInfoToCallerContext method to add clientIP and clientPort.

Reply:
Since it is redundant check, we can remove `ctx == null ? null :  ` from both L466-L467 correct?

Reply:
Yes, It can.  I moved them to a public location